conans/client/graph/graph_builder.py

```diff
@@ -123,13 +129,21 @@ def _load_deps(self, node, down_reqs, dep_graph, public_deps, down_ref, down_opt
                                     new_options, new_loop_ancestors, aliased, check_updates, update,
                                     remote_name, processed_profile)
 
+    @staticmethod
+    def _conflicting_references(ref1, ref2):
+        if ref1.copy_clear_rev() != ref2.copy_clear_rev():
+            return 1
+        if ref1.revision and ref2.revision and ref1.revision != ref2.revision:
```

Comment:
Actually `ref1.revision` shouldn't be `None` because that is the previous computed node. I'm changing some var names, introducing an assert and changing the comparison.
The idea is: The new revision (ref2.revision) can't be different from the previous one, but could be `None` without a conflict, when you write an override for example, `ref2` is the override and is not processed yet, and it shouldn't conflict with the previously resolved `ref1.revision`.

Reply:
Now is never None because it comes also from v1 with "0".
This is how it looks now:
```python
    @staticmethod
    def _conflicting_references(previous_ref, new_ref):
        if previous_ref.copy_clear_rev() != new_ref.copy_clear_rev():
            return REFERENCE_CONFLICT
        # Computed node, has to have a revision, at least 0
        assert(previous_ref.revision is not None)
        # If new_ref.revision is None we cannot assume any conflict, the user hasn't specified
        # a revision, so it's ok any previous_ref
        if new_ref.revision and previous_ref.revision != new_ref.revision:
            return REVISION_CONFLICT
        return False
```